scripts/generate_lms.py

```diff
@@ -0,0 +1,100 @@
+from collections import defaultdict
+import sys
+import os
+import logging
+from generator_utils import load_config, common_args, update_model
+
+
+logger = logging.getLogger(__name__)
+
+
+def create_organizations(config):
+    from common.djangoapps.util.organizations_helpers import add_organization
+
+    for code in config.get_organization_codes():
+        org = {
+            'short_name': code,
+            'name': config.get_organization_name(code)
+        }
+        logger.info('Creating Organization for {} - {}'.format(code, org))
+        add_organization(org)
+
+
+def create_sites(config):
+    from django.contrib.sites.models import Site
+
+    sites = {}
+    for code in config.get_microsite_codes():
+        context = config.get_context(code)
+        site = {
+            'domain': context['lms_domain'],
+            'name': context['lms_domain']
+        }
+        site = config.apply_overrides(code, 'lms', Site, site)
+        logger.info('Creating site for {} - {}'.format(code, site))
+        site, _ = Site.objects.get_or_create(domain=site['domain'], defaults=site)
+        sites[code] = site
+    return sites
+
+def create_site_configurations(config, sites):
+    from openedx.core.djangoapps.site_configuration.models import SiteConfiguration
+
+    for code in config.get_microsite_codes():
+        context = config.get_context(code)
+        site_configuration = {
+            'site': sites[code],
+            'enabled': True,
+            'site_values': {
+                'PLATFORM_NAME': context['name'],
+                'Platform_name': context['name'],
+                'LMS_ROOT_URL': context['lms_url'],
+                'LMS_BASE': context['lms_url'],
+                'ECOMMERCE_PUBLIC_URL_ROOT': context['ecommerce_url'],
+                'COURSE_CATALOG_API_URL': context['discovery_api_url'],
+            }
+        }
+        site_configuration = config.apply_overrides(code, 'lms', SiteConfiguration, site_configuration)
+        logger.info('Creating SiteConfiguration for {} - {}'.format(code, site_configuration))
+        sc, created = SiteConfiguration.objects.get_or_create(site=site_configuration['site'], defaults=site_configuration)
+        if not created:
+            update_model(sc, **site_configuration)
+
+
+def add_ecommerce_redirect_urls(config):
+    from oauth2_provider.models import Application
+
+    redirect_uris = []
+    for code in config.get_microsite_codes():
+        context = config.get_context(code)
+        redirect_uris.append('{}/complete/edx-oauth2/'.format(context['ecommerce_url']))
+
+    ecommerce_app = Application.objects.get(name='ecommerce-sso')
```

Comment:
What if this application is not existing, we should either through an error or probably should document this,

Reply:
In a correct installation, this application should exist. So we can assume it is there. I've left a comment about it and also moved the hardcoded `ecommerce-sso` to `const.py` file.